### DISCUSSION TITLE
How to Prevent Packages from Being Removed

### DISCUSSION THREAD

**Discussion by levone1** [syl20bnr/spacemacs#16073] [Q&A] [answered]
Been getting familiarized with emacs, but have a long way to go...  Very appreciative of Spacemacs' ease of use, etc.  One issue I'm having is not being able to figure out why certain packages are auto-removed at every startup, or how to prevent it.  I'll use Neotree as example.  'M-x install-package' > select Neotree > Return, causes successful installation, but then it is removed at the next startup.

Adding `(setq package-selected-packages '(neotree))` to init.el causes it to show on selected-packages, which makes a quick, easy install by 'M-x package-install-selected-packages', but still have to do it every time.

I found something somewhere that indicated I could add 
`(add-to-list 'load-path "/path/to/neotree")`
`(require 'neotree)`
but obviously that won't work if Neotree is no longer in the path, having been removed.

Is something not working right, or am I missing something?  How to "permanently" install it?

Thanks

**Comment by levone1** [comment]
So just found out that `dotspacemacs-additional-packages` does the trick, at least for neotree...  I did notice in the notes that putting package names there won't automatically load packages that are dependencies, and those would have to be listed there also.  What's the best way to find out what dependencies a given package needs?

Thanks

**Comment by dalanicolai** [comment, marked as answer]
Although this question is not about using neotree, just for completeness, for neotree there is also the [neotree layer](https://develop.spacemacs.org/layers/+filetree/neotree/README.html). Have not tested it though...

B.t.w., AFAIK, if a package is well written, then adding its name in `dotspacemacs-additional-packages` should also take care of (loading) dependencies. It is the responsibility of the package to 'declare' its dependencies via the dependencies package attribute (see [here](https://www.gnu.org/software/emacs/manual/html_node/elisp/Packaging-Basics.html)).

**Reply by levone1** [reply]
Quick additional semi-related question - by adding simply `(neotree)` to init.el, I can get Neotree to load when emacs starts, but is there a way I can get it to load with options?  For example, typing 'H' in the Neotree buffer causes hidden files to show, but can I have it start automatically showing hidden files?  I've tried changing `(neotree)` to `(neotree -H)` etc., just shooting in the dark...
